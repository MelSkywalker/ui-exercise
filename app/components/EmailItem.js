import React from 'react';
import dateFormatter from '../utils/date';
import messageFormatter from '../utils/text';
import ImportantIcon from './svg/ImportantIcon';

import './emailItem.scss';

const EmailItem = ({ content }) => {
	const { date, sender, subject, body, tags } = content;

	const formattedDate = dateFormatter(date);
	return (
		<div className="email-item">
			<input
				type="checkbox"
				className="msg-checkbox item-element"
			/>
			<button>
				<ImportantIcon className="item-element" />
			</button>
			<p className="sender item-element">{sender}</p>
			<p className="date item-element">{formattedDate}</p>
			<div className="message item-element">
				<span className="subject">{`${subject} - `}</span>
				<span className="message-body">{messageFormatter(body)}</span>
			</div>
			<div className="email-tags item-element">
				{tags && tags.map(tag => (
					<span key={tag} className="email-tag">{tag}</span>
				))}
			</div>
		</div>
	)
};

export default EmailItem;