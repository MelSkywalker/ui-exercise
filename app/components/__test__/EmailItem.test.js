import React from 'react';
import { shallow } from 'enzyme';

import EmailItem from '../EmailItem';

describe('components | EmailItem', () => {
	it('it renders component', () => {
		const props = {
			content: {
				subject: "hello",
				sender: "someone@fakemail.com",
				body: "Just wanted to say hello Just wanted to say hello Just wanted to say hello Just wanted to say hello.",
				date: "2017-03-05T10:25:43.511Z",
			}
		};

		const component = shallow(<EmailItem {...props} />);
		expect(component).toMatchSnapshot();
	})
})