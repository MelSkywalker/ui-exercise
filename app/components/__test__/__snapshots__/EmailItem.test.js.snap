// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | EmailItem it renders component 1`] = `
<div
  className="email-item"
>
  <input
    className="msg-checkbox item-element"
    type="checkbox"
  />
  <button>
    <ImportantIcon
      className="item-element"
    />
  </button>
  <p
    className="sender item-element"
  >
    someone@fakemail.com
  </p>
  <p
    className="date item-element"
  >
    3/5/17
  </p>
  <div
    className="message item-element"
  >
    <span
      className="subject"
    >
      hello - 
    </span>
    <span
      className="message-body"
    >
      Just wanted to say hello Just wanted to say hello Just wanted to say hello Just wanted to say hello.
    </span>
  </div>
  <div
    className="email-tags item-element"
  />
</div>
`;
